File: jiho/비밀지도.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<string> solution(int n, vector<int> arr1, vector<int> arr2) {
    vector<string> answer;
    for(int i=0;i<n;i++)
    {
        int result = arr1[i] | arr2[i];
        string str = "";
        for(int j=0;j<n;j++)
        {
            str += result % 2 == 1 ? "#" : " ";
            result /= 2;
        }
        reverse(str.begin(), str.end());
        answer.push_back(str);
    }
    return answer;
}

File: youngjun/week2/비밀지도.py
def solution(n, arr1, arr2):
    map1 = [[0 for j in range(n)] for i in range(n)]
    map2 = [[0 for j in range(n)] for i in range(n)]
    row = 0
    
    for i in range(n):
        row = format(arr1[i],'b')
        for j in range(len(row)):
            map1[i][n-1-j] = row[len(row)-j-1]
    
    for i in range(n):
        row = format(arr2[i],'b')
        for j in range(len(row)):
            map2[i][n-1-j] = row[len(row)-j-1]
    
    answer = []

    result = ''

    for k in range(n):
        for t in range(n):

            if int(map1[k][t]) == 1 or int(map2[k][t]) == 1:
                result+='#'
            else:
                result+=' '
        answer.append(result)
        result = ''
    
    
    return answer